### PR TITLE
3 bit descr

### LINKED ISSUE #3: Add BIT school data

### PR DESCRIPTION
**Pull request by Juan-Mateos**
Closes #3 

This PR includes:

* All the BIT raw data (`docx` files with computer science and geography descriptions by school)
* A script to parse these data into two dicts looking up school ids against compsci and geo descriptions  (NB I have also included the outputs in the `inputs/data` folder)
* Getters to read these lookups as well as the BIT master file looking up school ids with names and some other information
* Updated requirements (remember to `pip install -r requirements.txt --upgrade`) before trying to reproduce the code above.
---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [ ] I have requested a code review

### CODE REVIEW

File: comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import os
+import docx
+
+from comp_sci_gender_bias import PROJECT_DIR
+
+DATA_DIR = f"{PROJECT_DIR}/inputs/data/"
+BIT_DIRS = ["bit_compsci_descriptions", "bit_geo_descriptions"]
+
+
+def getText(filename: str) -> str:
```
Comment (Jack-0-0) [inline comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py]: Function `getText` should be in camel case: `get_text` as per the style guidelines

File: comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    """Reads and parses a doc file from BIT"""
+
+    doc = docx.Document(filename)
+    fullText = []
```
Comment (Jack-0-0) [inline comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py]: Variable `fullText` should be in camel case: `full_text` as per the style guidelines

File: comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    for para in doc.paragraphs:
+        fullText.append(para.text)
```
Comment (Jack-0-0) [inline comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py]: Creating `fullText` could be simplified to `fullText = [para.text for para in doc.paragraphs]`

File: comp_sci_gender_bias/getters/school_data.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+def school_table() -> pd.DataFrame:
+    """Reads a table with school metadata including lookup between ids and names"""
+
+    return pipe(
+        pd.read_csv(f"{PROJECT_DIR}/inputs/data/school_master_table.csv"),
+        lambda df: df.rename(
+            columns={col: re.sub(" ", "_", col.strip().lower()) for col in df.columns}
+        ),
+        lambda df: df.rename(columns={"number": "id"}),
```
Comment (Jack-0-0) [inline comp_sci_gender_bias/getters/school_data.py:31]: Nice use of `pipe` to rename the columns 🙂

**Review by Jack-0-0 — COMMENTED** [review]
Thanks for this PR @Juan-Mateos. All the code runs as expected 🙂

I have made a few comments on specific lines of code.

All the docx inputs are being stored on github. Why are we not storing these on S3?

**Comment by Juan-Mateos** [comment]
I addressed the comments above i.e. 
 - [ ] Changed `GetText` name
 - [ ] Removed `for` loop with list comprehension
 - [ ] Re-ran the script and confirmed it works

**Review by Jack-0-0 — APPROVED** [review]
Thanks @Juan-Mateos. This looks good to merge

**Pull request merged**